src/app/searchbox/searchbox.component.ts

```diff
@@ -0,0 +1,281 @@
+  //Update the number of people function
+  updateNumberOfPerson(person: string,event:any){
+    if (person == "Adult"){
+      let tempAdult = parseInt(event.target.value);
+      if (tempAdult < 0)
+      {
+        this.adult = 1;
+      }else{
+        this.adult = tempAdult;
+      }
+    }
+    if (person == "Children"){
+      let tempChildren = parseInt(event.target.value);
+      if (tempChildren < 0){
+        this.countOfChildren = 0;
+      }
+      else{
+        this.countOfChildren = tempChildren;
+      }
+    }
+    this.numberOfPeople = this.adult + " Adult";
+    if(this.countOfChildren > 0){
+      this.numberOfPeople += "; " + this.countOfChildren + " Children"; 
+    }
+  }
+
+  //Closing overlay after end user makes a choice
+  afterChoosingAnOption(overlay:string, choice:string, dataStore:string){
+    if ("tripRB" == dataStore){
+      this.tripRB = choice;
+      this.closeOverlay(overlay);
+      if(this.tripRB != "One-Way"){
+          this.disableToCalendar = false;
+          this.placeHolder = "To: yyyy-mm-dd";
+      }
+      else{
+        this.disableToCalendar = true;
+        this.toModel = undefined;
+        this.placeHolder = "No return date needed";
+      }
+      return;
+    }
+    if("nonStopRB" == dataStore){
+      this.nonStopRB = choice;
+      this.closeOverlay(overlay);
+      return;
+    }
+  }
+
+  //Toggle the overlay after button press
+  toggleOverlay(overlay: string, event: Event){
+    event.stopPropagation();
+
+    //Check there is another overlay up
+    if (this.overlay != ""){
+      this.closeOverlay(this.overlay);
+      this.overlay = "";
+    }
+
+    const overlayElement = document.getElementById(overlay);
+    if (overlayElement != null){
+      overlayElement.style.display = "initial";
+      this.overlay = overlay;
+    }
+  }
+
+  closeOverlay(overlay: string){
+    const overlayElement = document.getElementById(overlay);
+    if (overlayElement != null){
+      overlayElement.style.display = "none";
+      this.overlay = "";
+    }
+  }
+
+  //Set airport input from user
+  setInput(event: any, where:string){
+    let tempVal = event.target.value;
+    if (where == "from"){
+      this.fromAirport = tempVal;
+      return;
+    }
+    if (where == "to"){
+      this.toAirport = tempVal;
+    }
+  }
+
+  swapLocations(){
+    this.fromAirport = [this.toAirport, this.toAirport = this.fromAirport][0];
+  }
+
+  compareDates(date: NgbDateStruct | undefined, comparisonDate:NgbDateStruct | undefined){
+    if (date == undefined || comparisonDate == undefined){
+      return false;
+    }
+    if(date.year < comparisonDate.year){
+      return false;
+    }
+    if (date.year > comparisonDate.year){
+      return true;
+    }
+    if (date.month == comparisonDate.month){
+      if (date.day < comparisonDate.day){
+        return false;
+      }
+      return true;
+    }
+    if (date.month < comparisonDate.month){
+      return false;
+    }
+    return true;
+  }
+
+  search(){
```

Comment:
this is _really_ complicated and should probably be broken down into multiple functions or potentially modelled with a state machine